src/Servant/QuickCheck/Internal/HasGenRequest.hs

```diff
@@ -82,6 +82,16 @@ instance (Arbitrary c, HasGenRequest b, ToHttpApiData c )
         (oldf, old) = genRequest (Proxy :: Proxy b)
         new = arbitrary :: Gen c
 
+instance (Arbitrary a, HasGenRequest b, ToHttpApiData a)
+    => HasGenRequest (Fragment a :> b) where
+    genRequest _ = (oldf, do
+      old' <- old
+      new' <- toUrlPiece <$> new
+      return $ \burl -> let r = old' burl in r { path = path r <> cs new' })
```

Comment:
This will generate incorrect requests:

 - The fragment part of an URI is not meant to be included in an HTTP request (cf. [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986#section-3.5)).
 - The fragment has to be at the end of the URI, starting with a `#`.
 - In http-client, the `queryString` of a `Request` is appended to the `path`, so `new'` will end up *before* the query string.

Reply:
Indeed you are correct.
I forgot that the `Fragment` is never passed with HTTP request, and generally has spotty support for anything, but passing `@id` back from the server to the client.